com.unity.ml-agents/Runtime/Sensors/ITypedSensor.cs

```diff
@@ -0,0 +1,27 @@
+namespace Unity.MLAgents.Sensors
+{
+
+    /// <summary>
+    /// The ObservationType enum of the Sensor.
+    /// </summary>
+    internal enum ObservationType
+    {
+        Default = 0,
+        Goal = 1,
+        Reward = 2,
+        Message = 3,
+    }
+
+
+    /// <summary>
+    /// Sensor interface for sensors with variable types.
+    /// </summary>
+    internal interface ITypedSensor
```

Comment:
Shouldn't this interface be public? Are users expected to add `ObservationType` to their sensors or not for now?

Reply:
It doesn't need to be, since any ISensor that isn't an ITypedSensor should be treated as Default type.

Reply:
Intention is similar to making the DimensionProperty internal for now. To not change the public API until there are actual features to use it.